packages/dashboard/src/components/tour/tour.tsx

```diff
@@ -42,10 +39,26 @@ export const DashboardTour = React.memo(
       clearSpotlights,
       setCurrentView,
       doorSpotlight,
-      setDoorSpotlight,
     };
     const { tourSteps, theme } = createTourSteps(tourFunctions);
 
+    const [update, setUpdate] = React.useState(0);
+
+    /**
+     * Qn: Best solution to handle animations?
+     *
+     * 1. update the tour on an interval.
+     * 2. set a timeout between each type, this has the downside that it would look laggy and
+     * would also breaks when the animation length changes.
+     * 3. Listen for all animation end callback and update the tour. (current solution)
```

Comment:
Yeah, I think 3 is the best solution for the simple fact that we are relying on events and not just time-related-solutions.